docker/mongodb-kubernetes-tests/tests/search/search_enterprise_x509_cluster_auth.py

```diff
@@ -0,0 +1,261 @@
+
+    return resource
+
+
+@fixture(scope="function")
+def admin_user(namespace: str) -> MongoDBUser:
+    resource = MongoDBUser.from_yaml(
+        yaml_fixture("mongodbuser-mdb-admin.yaml"), namespace=namespace, name=ADMIN_USER_NAME
+    )
+
+    if try_load(resource):
+        return resource
+
+    resource["spec"]["mongodbResourceRef"]["name"] = MDB_RESOURCE_NAME
+    resource["spec"]["username"] = resource.name
+    resource["spec"]["passwordSecretKeyRef"]["name"] = f"{resource.name}-password"
+
+    return resource
+
+
+@fixture(scope="function")
+def user(namespace: str) -> MongoDBUser:
+    resource = MongoDBUser.from_yaml(yaml_fixture("mongodbuser-mdb-user.yaml"), namespace=namespace, name=USER_NAME)
+
+    if try_load(resource):
+        return resource
+
+    resource["spec"]["mongodbResourceRef"]["name"] = MDB_RESOURCE_NAME
+    resource["spec"]["username"] = resource.name
+    resource["spec"]["passwordSecretKeyRef"]["name"] = f"{resource.name}-password"
+
+    return resource
+
+
+@fixture(scope="function")
+def mongot_user(namespace: str, mdbs: MongoDBSearch) -> MongoDBUser:
+    resource = MongoDBUser.from_yaml(
+        yaml_fixture("mongodbuser-search-sync-source-user.yaml"),
+        namespace=namespace,
+        name=f"{mdbs.name}-{MONGOT_USER_NAME}",
+    )
+
+    if try_load(resource):
+        return resource
+
+    resource["spec"]["mongodbResourceRef"]["name"] = MDB_RESOURCE_NAME
+    resource["spec"]["username"] = MONGOT_USER_NAME
+    resource["spec"]["passwordSecretKeyRef"]["name"] = f"{resource.name}-password"
+
+    return resource
+
+
+@mark.e2e_search_enterprise_x509_cluster_auth
+def test_install_operator(namespace: str, operator_installation_config: dict[str, str]):
+    operator = get_default_operator(namespace, operator_installation_config=operator_installation_config)
+    operator.assert_is_running()
+
+
+@mark.e2e_search_enterprise_x509_cluster_auth
+@skip_if_cloud_manager
+def test_create_ops_manager(namespace: str):
+    ops_manager = get_ops_manager(namespace)
+    ops_manager.update()
+    ops_manager.om_status().assert_reaches_phase(Phase.Running, timeout=1200)
+    ops_manager.appdb_status().assert_reaches_phase(Phase.Running, timeout=600)
+
+
+@mark.e2e_search_enterprise_x509_cluster_auth
+def test_install_tls_secrets_and_configmaps(namespace: str, mdb: MongoDB, mdbs: MongoDBSearch, issuer: str):
+    create_agent_tls_certs(issuer, namespace, mdb.name, "certs")
+    create_x509_mongodb_tls_certs(issuer, namespace, mdb.name, f"certs-{mdb.name}-clusterfile")
+    create_x509_mongodb_tls_certs(issuer, namespace, mdb.name, f"certs-{mdb.name}-cert", mdb.get_members())
+
+    search_service_name = f"{mdbs.name}-search-svc"
+    create_tls_certs(
+        issuer,
+        namespace,
+        f"{mdbs.name}-search",
+        replicas=1,
+        service_name=search_service_name,
+        additional_domains=[f"{search_service_name}.{namespace}.svc.cluster.local"],
```

Comment:
Is this service FQDN actually validated by mongod when connecting to mongot? It should, right? 

Also when we switch to L7 our proxy component should probably present the same cert, or at least cert issued for the same service FQDN?

Reply:
@lsierant it does seem that when mongod connects to mongot the hostname is validated against mongot's TLS certificate. Commenting out the `additional_domains` specification here causes the `createSearchIndex` command to fail with a `Error connecting to Search Index Management service` message.